unity/Assets/Scripts/BaseFPSAgentController.cs

```diff
@@ -2053,7 +2087,7 @@ protected SimObjPhysics[] GetAllVisibleSimObjPhysics(Camera agentCamera, float m
             }
             #endif
 
-            HashSet<SimObjPhysics> currentlyVisibleItems = new HashSet<SimObjPhysics>();
+            List<SimObjPhysics> currentlyVisibleItems = new List<SimObjPhysics>();
```

Comment:
Why make this a list? Won't this make the `if (!currentlyVisibleItems.Contains(sop))` call below into a linear time op rather than constant? Is the idea that the number of visible items is generally small so this is, in practice, is faster than not having it be a hashset?

Reply:
All these changes are reverting back to the pre-simobject filter code to keep the visibility logic identical.

Reply:
Hard to say... I think based on the scene and the actual angle you are looking at objects, one or the other may have better performance just depending on the exact situation. Overall I can't say definitively if the average list length is small enough that it will perform better or not...